group/ristretto255.go

```diff
@@ -0,0 +1,179 @@
+package group
+
+import (
+	"crypto"
+	"io"
+
+	h2c "github.com/armfazh/h2c-go-ref"
+
+	r225 "github.com/bwesterb/go-ristretto"
+)
+
+var (
+	Ristretto255 Group = ristrettoGroup{}
+)
+
+type ristrettoGroup struct {
+}
+
+type ristrettoElement struct {
+	p *r225.Point
+}
+
+type ristrettoScalar struct {
+	s *r225.Scalar
+}
+
+func (g ristrettoGroup) NewElement() Element {
+	return &ristrettoElement{
+		p: &r225.Point{},
+	}
+}
+
+func (g ristrettoGroup) NewScalar() Scalar {
+	return &ristrettoScalar{
+		s: &r225.Scalar{},
+	}
+}
+
+func (g ristrettoGroup) Identity() Element {
+	var zero r225.Point
+	zero.SetZero()
+	return &ristrettoElement{
+		p: &zero,
+	}
+}
+
+func (g ristrettoGroup) Generator() Element {
+	var base r225.Point
+	base.SetBase()
+	return &ristrettoElement{
+		p: &base,
+	}
+}
+
+func (g ristrettoGroup) Order() Scalar {
+	panic("not implemented")
+	return nil
+}
+
+func (g ristrettoGroup) RandomElement(r io.Reader) Element {
+	var x r225.Point
+	x.Rand()
+	return &ristrettoElement{
+		p: &x,
+	}
+}
+
+func (g ristrettoGroup) RandomScalar(r io.Reader) Scalar {
+	var x r225.Scalar
+	x.Rand()
+	return &ristrettoScalar{
+		s: &x,
+	}
+}
+
+// Note(caw): this does NOT implement HashToElement as specified in the hash-to-curve draft
```

Comment:
I'm happy to make changes to go-ristretto to accommodate the spec.

Reply:
I think it's a new API. Derive is basically FromUniformBytes, but this is a wrapper around that interface.

Reply:
Could you be a bit more specific what functionality you need?